assets/src/edit-story/components/panels/link/index.js

```diff
@@ -99,15 +101,33 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
     if (!isValidUrl(urlWithProtocol)) {
       return;
     }
+    const invalidLinkMessage = __('This is an invalid link!', 'web-stories');
     setFetchingMetadata(true);
     getLinkMetadata(urlWithProtocol)
       .then(({ title, image }) => {
-        updateLinkFromMetadataApi({ url: urlWithProtocol, title, icon: image });
+        if (title && image) {
+          return updateLinkFromMetadataApi({
+            url: urlWithProtocol,
+            title,
+            icon: image,
+          });
+        }
+
+        // This is necessary since the API is returning empty image/title fields sometimes.
+        const invalidLinkError = new Error(invalidLinkMessage);
+        invalidLinkError.code = 'invalid_url';
```

Comment:
100% agree. I will remove this validation AND error `throw` from this `.then` statement since we e should report invalid URLs before or in the `.catch` statement. Sometimes the URL will be valid, but will not return image/title, so the elaborated logic is not correct.